### PR TITLE
Feature/google oauth

### PR DESCRIPTION
**Pull request by ayushkamadji**
Initial PR for #512 

*Important note: need to config api id/secret*
*More notes:*
*- the view/styling is basic since there isn't a mock for it yet*
*- omniauth callback controller can be refactored*

*EDIT:* 
##### Login
screen shot 2017-07-28 at 09 36 22-fullpage

##### Sign up
screen shot 2017-07-28 at 09 35 57-fullpage

### CODE REVIEW

File: app/assets/stylesheets/components/login_page.scss [hunk 1]
```diff
@@ -105,7 +105,16 @@
     background: $text-primary;
     padding-top: 14.2px;
     vertical-align: middle;
+    margin-bottom: 10px;
   }
+
+  .google-login-button {
+    background: $google-color;
+    padding-top: 14.2px;
```
Comment (KevinMulhern) [inline app/assets/stylesheets/components/login_page.scss]: can we create a new class to hold the duplication between the two omniauth buttons please?

File: app/assets/stylesheets/settings.scss [hunk 2]
```diff
@@ -30,3 +30,6 @@ $whitish-grey: #d8d8d8;
 //buttons
 $button-primary: #6abfc3;
 $button-seconday: #cc9543;
+
+//3rd party brands
+$google-color: #dd4b39;
```
Comment (KevinMulhern) [inline app/assets/stylesheets/settings.scss]: We will need to check with @adachiu with this color. We could possibly go with googles blue color.

File: app/views/shared/_github_button.html.erb [hunk 3]
```diff
@@ -1,9 +1,9 @@
 <% if @user.password_required? %>
   <% resource_class.omniauth_providers.each do |provider| %>
     <div class="omniauth-login text-center">
-      <%= link_to omniauth_authorize_path(resource_name, provider), class: 'github-login-button page-buttons login btn' do %>
```
Comment (KevinMulhern) [inline app/views/shared/_github_button.html.erb]: Now that this partial contains more than one button, can we rename it to `_omniauth_buttons.html.erb` please?

File: app/controllers/omniauth_callbacks_controller.rb [hunk 4]
```diff
@@ -18,6 +18,22 @@ def github
     end
   end
 
+  def google
```
Comment (KevinMulhern) [inline app/controllers/omniauth_callbacks_controller.rb]: Can we add unit tests for this method please? they will be pretty much the same as the ones for the `github` method

**Comment by KevinMulhern** [comment]
Hi @ayushkamadji, thank you for submitting this PR 😄 It's looking really good. I've left a few minor comments, let me know if you want any clarification on any of them.

Can you add a screen shot of the new omniauth buttons to the description of this PR as well please? 😄

File: app/views/shared/_github_button.html.erb (hunk 3 above)
Comment (ayushkamadji) [inline app/views/shared/_github_button.html.erb]: Just to be sure, this partial is only called from the login and sign up pages yes? Can anyone confirm?

File: app/assets/stylesheets/settings.scss (hunk 2 above)
Comment (ayushkamadji) [inline app/assets/stylesheets/settings.scss]: Sure, I'll leave it for now as place holder until @adachiu has reviewed

File: app/views/shared/_github_button.html.erb (hunk 3 above)
Comment (KevinMulhern) [inline app/views/shared/_github_button.html.erb]: Yeah those are the only places its called from.

**Comment by KevinMulhern** [comment]
Thanks for submitting those changes @ayushkamadji. Thinking about this a bit more, theres a lot of duplication between the `github` and `google` methods in the omniauth callbacks controller. 

We can combine the two methods into one, that will reduce the amount of code we have to maintain in the long run and negate the need to have duplicate tests as well.

But I'll leave that as a follow up refactor task for this feature and won't hold this PR from being merged.

Thanks for your work on this 👍 LGTM

**Pull request merged**